File: Mobile/lib/shared/utils/status_color.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_colors.dart';

({Color bg, Color text, Color border}) getStatusColors(String status) {
  return switch (status) {
    'Created' => (bg: AppColors.statusCreatedBg, text: AppColors.statusCreatedText, border: AppColors.statusCreatedBorder),
    'Received at Warehouse' => (bg: AppColors.statusReceivedBg, text: AppColors.statusReceivedText, border: AppColors.statusReceivedBorder),
    'Assigned to Linehaul' => (bg: AppColors.statusLinehaulBg, text: AppColors.statusLinehaulText, border: AppColors.statusLinehaulBorder),
    'Picked Up' => (bg: AppColors.statusPickedUpBg, text: AppColors.statusPickedUpText, border: AppColors.statusPickedUpBorder),
    'In Transit' => (bg: AppColors.statusTransitBg, text: AppColors.statusTransitText, border: AppColors.statusTransitBorder),
    'Arrived at Warehouse' => (bg: AppColors.statusArrivedBg, text: AppColors.statusArrivedText, border: AppColors.statusArrivedBorder),
    'Assigned to Courier' => (bg: AppColors.statusCourierBg, text: AppColors.statusCourierText, border: AppColors.statusCourierBorder),
    'Out For Delivery' => (bg: AppColors.statusOutDelivBg, text: AppColors.statusOutDelivText, border: AppColors.statusOutDelivBorder),
    'Delivered' => (bg: AppColors.statusDeliveredBg, text: AppColors.statusDeliveredText, border: AppColors.statusDeliveredBorder),
    'Failed Delivery' => (bg: AppColors.statusFailedBg, text: AppColors.statusFailedText, border: AppColors.statusFailedBorder),
    _ => (bg: AppColors.statusCreatedBg, text: AppColors.statusCreatedText, border: AppColors.statusCreatedBorder),
  };
}

File: Frontend/src/utils/statusColor.js
import { PACKAGE_STATUS } from '../constants/packageStatus';

export function statusColor(status) {
  switch (status) {
    case PACKAGE_STATUS.CREATED:
      return 'bg-status-created-bg text-status-created-text border-status-created-border';
    case PACKAGE_STATUS.RECEIVED_AT_WAREHOUSE:
      return 'bg-status-received-bg text-status-received-text border-status-received-border';
    case PACKAGE_STATUS.ASSIGNED_TO_LINEHAUL:
      return 'bg-status-linehaul-bg text-status-linehaul-text border-status-linehaul-border';
    case PACKAGE_STATUS.PICKED_UP:
      return 'bg-status-pickedup-bg text-status-pickedup-text border-status-pickedup-border';
    case PACKAGE_STATUS.IN_TRANSIT:
      return 'bg-status-transit-bg text-status-transit-text border-status-transit-border';
    case PACKAGE_STATUS.ARRIVED_AT_WAREHOUSE:
      return 'bg-status-arrived-bg text-status-arrived-text border-status-arrived-border';
    case PACKAGE_STATUS.ASSIGNED_TO_COURIER:
      return 'bg-status-courier-bg text-status-courier-text border-status-courier-border';
    case PACKAGE_STATUS.OUT_FOR_DELIVERY:
      return 'bg-status-outdelivery-bg text-status-outdelivery-text border-status-outdelivery-border';
    case PACKAGE_STATUS.DELIVERED:
      return 'bg-status-delivered-bg text-status-delivered-text border-status-delivered-border';
    case PACKAGE_STATUS.FAILED_DELIVERY:
      return 'bg-status-failed-bg text-status-failed-text border-status-failed-border';
    default:
      return 'bg-gray-100 text-gray-800 border-gray-300';
  }
}
